internal/pkg/test/tool/require/require_linux.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package require
+
+import (
+	"fmt"
+	"io"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"testing"
+
+	"github.com/sylabs/singularity/internal/pkg/buildcfg"
+	"github.com/sylabs/singularity/pkg/network"
+)
+
+var hasUserNamespace bool
+var hasUserNamespaceOnce sync.Once
+
+var supportNetwork bool
+var supportNetworkOnce sync.Once
+
+// UserNamespace checks that the current test could use
+// user namespace, if user namespaces are not enabled or
+// supported, the current test is skipped with a message.
+func UserNamespace(t *testing.T) {
+	// not performance critical, just save extra execution
+	// to get the same result
+	hasUserNamespaceOnce.Do(func() {
+		// user namespace is a bit special, as there is no simple
+		// way to detect if it's supported or enabled via a call
+		// on /proc/self/ns/user, the easiest and reliable way seems
+		// to directly execute a command by requesting user namespace
+		cmd := exec.Command("/bin/true")
+		cmd.SysProcAttr = &syscall.SysProcAttr{
+			Cloneflags: syscall.CLONE_NEWUSER,
+		}
+		// no error means user namespaces are enabled
+		hasUserNamespace = cmd.Run() == nil
+	})
+	if !hasUserNamespace {
+		t.Skipf("user namespaces seems not enabled or supported")
+	}
+}
+
+// Network check that bridge network is supported by
+// system, if not the current test is skipped with a
+// message.
+func Network(t *testing.T) {
+	supportNetworkOnce.Do(func() {
+		var err error
+		var cmdPath string
+		var stdinPipe io.WriteCloser
+
+		cmdPath, err = exec.LookPath("cat")
+		if err != nil {
+			t.Fatal(err)
+		}
+		cmd := exec.Command(cmdPath)
+		cmd.SysProcAttr = &syscall.SysProcAttr{}
+		cmd.SysProcAttr.Cloneflags = syscall.CLONE_NEWNET
+
+		stdinPipe, err = cmd.StdinPipe()
+		if err != nil {
+			return
```

Comment:
Could we return the error here instead of nothing and print that as the reason the bridge network is not supported?

Reply:
Will log error but can't change function prototype, all functions in `require` must have the same prototype as test function where any errors -> skip current test